source/module_esolver/esolver_ks_pw.cpp

```diff
@@ -210,7 +210,7 @@ void ESolver_KS_PW<T, Device>::before_scf(const int istep)
     //----------------------------------------------------------
     // about vdw, jiyy add vdwd3 and linpz add vdwd2
     //----------------------------------------------------------
-    auto vdw_solver = vdw::make_vdw(GlobalC::ucell, PARAM.inp);
+    auto vdw_solver = vdw::make_vdw(GlobalC::ucell, PARAM.inp, &(GlobalV::ofs_running));
```

Comment:
can we pass a variable? why do we need passing an address?

Reply:
@mohanchen this is for implementing an optional logging via:
```c++
std::unique_ptr<vdw> make_vdw(const UnitCell& ucell, const Input& param, std::ofstream* plog = nullptr);
```
once the logging is not needed, just call with the former two parameters specified, see:
https://github.com/deepmodeling/abacus-develop/blob/edfa4c120b4edfb012d23fd1e16ef92cb868aaac/source/module_hamilt_pw/hamilt_pwdft/forces.cpp#L182
similar implementation can also be found here:
https://github.com/deepmodeling/abacus-develop/blob/edfa4c120b4edfb012d23fd1e16ef92cb868aaac/source/module_basis/module_nao/atomic_radials.cpp#L24-L59